ee/watchdog/controller_windows.go

```diff
@@ -180,224 +191,426 @@ func (wc *WatchdogController) ServiceEnabledChanged(enabled bool) {
-		}
+	taskName := launcher.TaskName(identifier, watchdogTaskType)
+	// init COM
+	ole.CoInitialize(0)
+	defer ole.CoUninitialize()
 
-		wc.slogger.Log(ctx, slog.LevelInfo, "removed watchdog service")
+	// create our scheduler object
+	schedService, err := oleutil.CreateObject("Schedule.Service")
+	if err != nil {
+		return fmt.Errorf("creating schedule service object: %w", err)
+	}
+	defer schedService.Release()
 
-		return
+	// get service interface handle
+	scheduler, err := schedService.QueryInterface(ole.IID_IDispatch)
+	if err != nil {
+		return err
 	}
+	defer scheduler.Release()
 
-	// we're enabling the watchdog - first check if we've already installed the service
-	// there are three potential paths here:
-	// 1. service did not previously exist, proceed with clean installation
-	existingService, err := serviceManager.OpenService(launcherWatchdogServiceName)
-	if err != nil && err.Error() == serviceDoesNotExistError {
-		if err = wc.installService(serviceManager); err != nil {
-			wc.slogger.Log(ctx, slog.LevelError,
-				"encountered error installing watchdog service",
-				"err", err,
-			)
-		}
+	// connect to the scheduler handle on the local machine
+	_, err = oleutil.CallMethod(scheduler, "Connect")
+	if err != nil {
+		return fmt.Errorf("failed to connect to Task Scheduler: %w", err)
+	}
 
-		return
+	// grab the root task folder
+	rootFolderVar, err := oleutil.CallMethod(scheduler, "GetFolder", `\`)
+	if err != nil {
+		return fmt.Errorf("failed to get root folder: %w", err)
 	}
 
-	// 2. we are unable to check the current status of the service,
-	// this is the least likely option and there's nothing we can do here so log and return
+	rootFolder := rootFolderVar.ToIDispatch()
+	defer rootFolder.Release()
+
+	// begin definition for a new task
+	taskDefinitionTemplate, err := oleutil.CallMethod(scheduler, "NewTask", 0)
 	if err != nil {
-		wc.slogger.Log(ctx, slog.LevelWarn,
-			"encountered error checking for watchdog service, unable to proceed with enabling",
-			"err", err,
-		)
+		return fmt.Errorf("failed to create new task definition: %w", err)
+	}
 
-		return
+	taskDefinition := taskDefinitionTemplate.ToIDispatch()
+	defer taskDefinition.Release()
+
+	installationDate := time.Now().Format(taskDateFormat)
+
+	// get the task registration info props
+	regInfoProp, err := oleutil.GetProperty(taskDefinition, "RegistrationInfo")
+	if err != nil {
+		return fmt.Errorf("getting registration info property: %w", err)
 	}
 
-	// 3. The watchdog service already exists on this device. Here we just restart it to ensure it is
-	// running on the latest launcher code
-	defer existingService.Close()
-	if err = wc.restartService(existingService); err != nil {
-		wc.slogger.Log(ctx, slog.LevelError,
-			"failure attempting to restart watchdog service",
-			"err", err,
-		)
+	regInfo := regInfoProp.ToIDispatch()
+	defer regInfo.Release()
+
+	if _, err = oleutil.PutProperty(regInfo, "Description", "Kolide agent waker"); err != nil {
```

Comment:
Instead of "waker" should we use "watchdog" for consistent terminology throughout codebase? Or maybe "restarter" for better precision?

Reply:
oh yep! restarter sounds good to me, the task name should include watchdog already. thank you!